Frontend Modularization: Datasets

itsmo-amzn opened the pull request:
### Feature or Bugfix
<!-- please choose -->
- Refactoring

### Detail
- Move datasets views into dataset module
- Move services that were only used by datasets into datasets module
- Move datasets components into datasets module

### Relates
- #398 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

nikpodsh commented on frontend/src/modules/Datasets/views/DatasetView.js:
```diff
@@ -31,25 +31,21 @@ import {
   useSettings
-} from '../../../../design';
-import { SET_ERROR, useDispatch } from '../../../../globalErrors';
+} from '../../../design';
```
routes.js should be fixed to match new paths for all view

nikpodsh commented on frontend/src/modules/Datasets/views/DatasetCreateForm.js:
```diff
@@ -29,15 +29,15 @@ import {
   useSettings
-} from '../../../../design';
-import { SET_ERROR, useDispatch } from '../../../../globalErrors';
+} from '../../../design';
```
Datasets (and everything related to it as tables, locations/folders, shares) is one of the optional modules. The views should not be shown if the module is disabled. We should do the same as it has been done for notebooks

itsmo-amzn replied:
can you please elaborate?

nikpodsh replied:
We shouldn't show dataset's forms if the dataset module is inactive. It was one of the requirements of the modularization: s customer can specify what modules are active/inactive in the config.json. 
The other example of the module that we can make inactive is `notebook`. If we set `modules.notebook.active = false` in the config.json, we won't see anything related to notebooks. For the reference, please take a look at #415. 
The list of the optional [modules](https://github.com/awslabs/aws-dataall/blob/modularization-main/config.json)

itsmo-amzn commented:
@jorgeig-space, and @nikpodsh 
I moved all shared components into the Shared module, and shared services into services module.
The PR is quite big as it affects many other modules that are pending modularization.

I also updated the datasets module to be optional in case the user disables it in the configs.json file

nikpodsh commented on frontend/src/services/graphql/Datasets/updateDatasetStorageLocation.js:
I wonder how we are going to deal with imports for all dataset related enitities like Folders and Tables ? e.g. `FolderEditForm` needs this `updateDatasetStorageLocation.js`, but `FolderEditForm`  will end up in `Folder` module

I think we need to discuss:
1) if we are going to allow importing from one module to another e.g. `FolderCreationForm` will have `import {addDatasetStorageLocation} from ../../../Datasets/services`. 
2) if we are going to put **shared for Dataset & co only**  services to the Shared folder
3) Come up with a bit different layout for datasets

nikpodsh commented on frontend/src/services/graphql/Datasets/index.js:
```diff
@@ -0,0 +1,8 @@
+export * from './addDatasetStorageLocation';
```
All imports for Folder's view should be fixed according to a new location of services

nikpodsh commented on frontend/src/modules/Shared/index.js:
```diff
@@ -0,0 +1,9 @@
+export * from './FeedComments';
```
Is it a comprehensive list of all `Shared` components or is it just what has been found so far? 
I am asking because if the list keeps growing (in the scope of modularization) we might need to think about reorganizing this into something like 
```
    Shared
           Comments
                  FeedCommentAdd
                  FeedComments
           Stack
                  Stack Stuff
           KeyValueTags
                  Key Value Stuff etc
```
And import it as `import { FeedComments } from 'modules/Shared/Comments';`
But it don't expect anything to pop up we can proceed

nikpodsh commented on frontend/src/modules/Shared/index.js:
```diff
@@ -0,0 +1,9 @@
+export * from './FeedCommentAdd';
+export * from './KeyValueTagList';
+export * from './KeyValueTagUpdateForm';
+export * from './ShareInboxList';
```
`ShareInboxList` and `ShareInboxListItem` are pure dataset functionality. If datasets are off `listDatasetShareObjects` and `getShareRequestsToMe` will fail

nikpodsh replied:
It works for me. Let's leave it there for a while. But we should not forget to return back to it when we modularize dataset shares.

nikpodsh commented on frontend/src/modules/Datasets/services/index.js:
```diff
@@ -1,19 +1,11 @@
-export * from './getDataset';
-export * from './getDatasetAssumeRoleUrl';
 export * from './getDatasetPresignedUrl';
 export * from './getDatasetSchema';
```
`getDatasetSchema` is no longer used and can be deleted as far as I see

itsmo-amzn commented on frontend/src/modules/Datasets/services/index.js:
```diff
@@ -1,19 +1,11 @@
-export * from './addDatasetStorageLocation';
 export * from './createDataset';
```
Sure, will bring this up during the next standup.
